### PR TITLE
ply: fix switching dataset error

### PR DESCRIPTION
**Pull request by belom88**

### CODE REVIEW

File: examples/website/pointcloud/app.js [hunk 1]
```diff
@@ -123,7 +123,7 @@ export default class App extends PureComponent {
   _onLoad({header, loaderData, attributes, progress}) {
     // metadata from LAZ file header
     const {maxs, mins} =
-      loaderData.header.mins && loaderData.header.maxs
+      loaderData.header && loaderData.header.mins && loaderData.header.maxs
```
Comment (ibgreen) [inline examples/website/pointcloud/app.js:126]:
```suggestion
    loaderData?.header.mins && loaderData?.header.maxs
```
Maybe we can use `mesh.header.boundingBox?`

File: modules/ply/src/lib/parse-ply.ts [hunk 2]
```diff
@@ -69,17 +68,31 @@ function parseHeader(data: any, options: {[index: string]: any}): PlyHeader {
     headerText = result[1];
     headerLength = result[0].length;
   }
+  const lines = headerText.split('\n');
+  const header = parseHeaderLines(lines, headerLength, options);
 
+  return header;
+}
+/**
```
Comment (ibgreen) [inline modules/ply/src/lib/parse-ply.ts:76]: Nit: add empty line

**Review by ibgreen — APPROVED** [review]

File: examples/website/pointcloud/app.js (hunk 1 above)
Comment (belom88) [reply examples/website/pointcloud/app.js:126]: The issue here occurs on the draco dataset. It doesn't contain "header"

Comment (ibgreen) [reply examples/website/pointcloud/app.js:126]: I see. The `loaderData` is loader specific and should not be used for generic processing. I have a PR up that fixes the mesh typing.

Comment (ibgreen) [reply examples/website/pointcloud/app.js:126]: But for a fix on release branch this is OK.

Comment (belom88) [reply examples/website/pointcloud/app.js]: Done

File: modules/ply/src/lib/parse-ply.ts (hunk 2 above)
Comment (belom88) [reply modules/ply/src/lib/parse-ply.ts:78]: Done

**Pull request merged**